pkg/retry/retry.go

```diff
@@ -58,6 +67,12 @@ func Do(fn func() error, opts ...Option) error {
 			return err
 		}
 
+		if cfg.backoffMultiplier != 0 {
+			interval := time.Duration(backoff) * cfg.initialInterval
+			backoff *= cfg.backoffMultiplier
```

Comment:
Nit: Looks strange that you're setting `backoff` after using it.

Reply:
How's so? I'm updating it's value for the next iteration.